bftengine/tests/testMsgsCertificate/msgsCertificate_test.cpp

```diff
@@ -1,10 +1,132 @@
+#include <messages/CheckpointMsg.hpp>
 #include "gtest/gtest.h"
+#include "SimpleClientImp.cpp"
+#include "messages/MsgsCertificate.hpp"
+#include "messages/ClientReplyMsg.hpp"
+#include "CheckpointInfo.hpp"
+#include "Digest.hpp"
 
-using namespace std;
+using namespace bftEngine;
 
-TEST(msgsCertificate_test, initial_test) {}
+namespace {
+
+const uint16_t fval = 1;
+const uint16_t cval = 1;
+const uint16_t numOfReplicas = 3 * fval + 2 * cval + 1;
+const uint16_t numOfRequired = 2 * fval + cval + 1;
+const uint16_t selfReplicaId = 0;
+char* reply = new char[10]{};
+int replyLen = strlen(reply);
+
+ReplicaId randomReplicaeId() { return rand() % numOfReplicas; }
+
+class mockCheckpointMsgCmp : public CheckpointInfo {
+ public:
+  static bool equivalent(CheckpointMsg* a, CheckpointMsg* b) {
+    return CheckpointInfo::CheckpointMsgCmp::equivalent(a, b);
+  }
+};
+
+TEST(msgsCertificate_test, obtain_inconsistent_state) {
+  MsgsCertificate<ClientReplyMsg, false, false, true, SimpleClientImp> replysCertificate(
+      numOfReplicas, fval, numOfRequired, selfReplicaId);
+
+  uint8_t reqSeqNum = 0;
+
+  // Add bunch of new messages
+  for (; reqSeqNum < 3; reqSeqNum++) {
+    ReplicaId replicaId = randomReplicaeId();
+    auto newMsg = new ClientReplyMsg(replicaId, reqSeqNum, reply, replyLen);
+    replysCertificate.addMsg(newMsg, replicaId);
+  }
```

Comment:
I think the "replicaId" shouldn't be drawn from the RNG. If I understand correct we want different replica ID to add in every round.